File: src/main/java/com/range/rpms/user/service/impl/UserServiceImpl.java
package com.range.rpms.user.service.impl;

import com.range.rpms.user.domain.model.User;
import com.range.rpms.user.domain.repository.UserRepository;
import com.range.rpms.user.exception.UserNotFoundException;
import com.range.rpms.user.service.UserService;
import org.springframework.stereotype.Service;

@Service
public class UserServiceImpl implements UserService {
    private final UserRepository userRepository;
    public UserServiceImpl(UserRepository userRepository) {
        this.userRepository = userRepository;
    }



    @Override
    public User findUser(Long id) {
     return userRepository.findById(id).orElseThrow((()->
             new UserNotFoundException("User not found")));
    }

    @Override
    public User findUserByUsername(String username) {
        return userRepository.findByUsername(username).orElseThrow(()->
                new UserNotFoundException("User not found"));
    }
}

File: src/main/kotlin/com/example/rpms/users/service/impl/UserServiceImpl.kt
package com.example.rpms.users.service.impl

import com.example.rpms.users.domain.entity.User
import com.example.rpms.users.domain.repository.UserRepository
import com.example.rpms.users.exception.UserNotFoundException
import com.example.rpms.users.service.UserService
import org.springframework.stereotype.Service

@Service
class UserServiceImpl (
    private val userRepository: UserRepository,
): UserService {
    override fun findByUsername(username: String): User {
        return userRepository.findByUsername(username)?:throw UserNotFoundException("")
    }

    override fun findByEmail(email: String): User {
        TODO("Not yet implemented")
    }

}